src/attr/_make.py

```diff
@@ -1562,6 +1576,9 @@ def wrap(cls):
                     " init must be True."
                 )
 
+        if match_args and "__match_args__" not in cls.__dict__:
```

Comment:
Please use [`_has_own_attribute`](https://github.com/python-attrs/attrs/blob/20c2d4fc0c72a844e12899d5d5e0adbc4741ce2c/src/attr/_make.py#L403) for the second check.

I'm aware that the implementation is currently obtuse and depending on how Larry's typing proposal will go, we'll fix it centrally eventually.

Reply:
This behavior is also somewhat inconsistent with the rest of `attrs`, where we overwrite unless `auto_detect` is set to True (default in new APIs). Is this the behavior of dataclasses?

Reply:
It also doesn't make sense to add it on Pythons older than 3.10, no?

Reply:
In the initial implementation of the pattern matching PEP there was no flag to disable `__match_args__` generation but there was a request raised later on a flag to opt-out though `__match_args__ = ()` being set manually was proposed. Discussion : https://bugs.python.org/issue43764